Main.hs

```diff
@@ -92,14 +102,30 @@ main = Rib.run [reldir|src|] [reldir|dist|] generateSite
       -- Copy over the static files
       Rib.buildStaticFiles [[relfile|static/**|]]
       -- Build individual markup sources, generating .html for each.
-      docs <- Rib.buildHtmlMulti patterns $ renderPage . Page_Doc
+      docs <- buildHtmlMulti'
+        MMark.parse
+        [[relfile|**/*.md|]]
+        (renderPage . Page_Doc)
 
       -- Build an index.html linking to the aforementioned files.
-      Rib.buildHtml [relfile|index.html|] $ renderPage $ Page_Index docs
-
-    -- File patterns to build, using the associated markup parser
-    patterns = Map.fromList [ ([relfile|*/*.md|], Some Rib.Markup_MMark)]
-
+      Rib.writeHtml [relfile|index.html|] $ renderPage $ Page_Index docs
+
+    buildHtmlMulti' :: Rib.SourceReader repr -> [Path Rel File] -> (Source repr -> Html ()) -> Action [Source repr]
```

Comment:
I'll think of ways to improve the next version of Rib to customize the target HTML file's URL so we don't have to 'fork' this function here like this, but meanwhile this should work.